# <center> Лабораторная работа № 1</center>
Перед началом выполнения каждой лабораторной работы, необходимо выполнить код в двух ячейках ниже. В случае если модуль `datetime` не установлен нужно установить. В последнем принте, вывести свою фамилию и инициалы.

In [1]:
!whoami

nevseros\ҐЄ вҐаЁ­ 


In [2]:
from datetime import datetime

current_time = datetime.now()
print(current_time)
print("Kaneva E.P.") # написать здесь свою фамилию и инициалы

2022-11-29 00:31:10.641188
Kaneva E.P.


# Задания

**Задание 1.** Программа должна заращивать у пользователя ввести любое положительное число и вывести первую и последню цифру этого числа, кол-во цифр в числе, сумму цифр в числе.

In [3]:
import time

print("Input positive number: ", end = "")
time.sleep(0.5) # а то он запрос раньше сообщения выводит
number = float(input())

# check
while number < 0:
    print("Try again")
    print("Input positive number: ", end = "")
    number = float(input())

# 1st digit
a = str(number)
print("1st digit:", int(a[0]))

# last digit
print("Last digit:", int(a[-1]))

# number of digits
if number == int(number):
    print("Number of digits:", len(str(number)))
else:                                                   # если числа целые, то
    print("Number of digits:", len(str(number)) - 1)    # эта ветка else не нужна

# sum
sumOfDigits = 0
if number != int(number):
    a = number % 1
    len_a = len(str(a)) - 2
    a = int(a * (10 ** len_a))
    digits_a = [int(digit) for digit in list(str(a))]
else:
    digits_a = [0]
b = int(number)
digits_b = [int(digit) for digit in list(str(b))]
sumOfDigits = sum(digits_a) + sum(digits_b)
print("Sum of digits:", sumOfDigits)

# если всё целое, то сумму можно реализовать так:
# digits = [int(digit) for digit in list(str(number))]
# print("Sum of digits:", sum(digits))

Input positive number: 0.123456
1st digit: 0
Last digit: 6
Number of digits: 7
Sum of digits: 21


**Задание 2.** Программа должна заращивать у пользователя ввести любое  положительное пятизначное число и вывести третью цифру этого числа.

In [4]:
from time import sleep

# input
print("Input positive whole number: ", end = "")
time.sleep(0.5) # а то он запрос раньше сообщения выводит
number = float(input())

# check
while number < 0 or number != int(number):
    print("Try again")
    print("Input positive whole number: ", end = "")
    number = float(input())

# main
number = str(number)
print("3rd digit:", number[2])

Input positive whole number: 12345
3rd digit: 3


**Задание 3.** Определить. 
Программа должна заращивать у пользователя ввести любое два `a` и `b` положительных числа.
Затем необходимо вычислить выражения `a//-b` и `a%-b`, а после объяснить (в ячейке *markdown*) как происходит вычисления в подобных выражениях.

In [5]:
# input
a = int(input("Input positive whole number a: "))
b = int(input("Input positive whole number b: "))

# check
while a < 0 or b < 0:
    print("Try again")
    a = int(input("Input positive whole number a: "))
    b = int(input("Input positive whole number b: "))

# main
print("a // -b =", a // (-b), "\na % -b =", a % (-b))

Input positive whole number a: 1234
Input positive whole number b: 123
a // -b = -11 
a % -b = -119


При делении положительного числа на отрицательное остаток отрицателен, т.е. `1234 % -123 = -119`, т.к. `1234 = -11 * -123 + (-119)`.

**Задание 4.** Напишите, программу, чтобы текст выводился в таком виде, как представлен ниже. Программа должна запращивать два значения размер партии и кол-во бракованных единиц `batch_size, number_of_defective`, и посчитать долю брака в процентах `percentage_of_defective`. Если процент брака больше 15, тогда напечатать слово **ужасны**, вместо **прекрасны** (вместо **почти нет** на **много**).


    Хотим сообщить, что ваши ноутбуки прекрасны(ужасны) и брака почти нет (много).
    Из партии, которую вы нам прислали, в размере
    batch_size
    единиц, всего
    number_of_defective
    ноутбуков были бракованными. Что составляет около
    percentage_of_defective %. 

    С уважением и любовью,
    Иван Иванович
    тел. +7234567891

In [6]:
batch_size = int(input("Введите размер партии: "))
number_of_defective = int(input("Введите количество брака: "))
percentage_of_defective = (number_of_defective / batch_size) * 100

if percentage_of_defective > 15:
    insert_1 = "ужасны"
    insert_2 = "много"
else:
    insert_1 = "прекрасны"
    insert_2 = "почти нет"
    
print(
f"""Хотим сообщить, что ваши ноутбуки {insert_1} и брака {insert_2}.
Из партии, которую вы нам прислали, в размере
{batch_size}
единиц, всего
{number_of_defective}
ноутбуков были бракованными. Что составляет около
{percentage_of_defective} %.\n
С уважением и любовью,
Иван Иванович
тел. +7234567891"""
)

Введите размер партии: 123456
Введите количество брака: 123
Хотим сообщить, что ваши ноутбуки прекрасны и брака почти нет.
Из партии, которую вы нам прислали, в размере
123456
единиц, всего
123
ноутбуков были бракованными. Что составляет около
0.09963063763608088 %.

С уважением и любовью,
Иван Иванович
тел. +7234567891
